### PR TITLE
done

### PR DESCRIPTION
**Pull request by AndriiYelieva**
[DEMO LINK](https://AndriiYelieva.github.io/react_movies-list-add-form/)

### CODE REVIEW

File: src/components/NewMovie/NewMovie.tsx [hunk 1]
```diff
@@ -1,45 +1,94 @@
...
+  const [movie, setMovie] = useState(initialMovieState);
+
+  const hasInvalidFields = !movie.title || !movie.imgUrl
+    || !movie.imdbUrl || !movie.imdbId;
+
+  const resetForm = () => {
+    setMovie(initialMovieState);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
```
Comment (Viktor-Kost) [inline src/components/NewMovie/NewMovie.tsx:28]: The `handleSubmit` method should validate if the URLs for `imgUrl` and `imdbUrl` are valid using a regex pattern. This is a part of the task's optional requirements.

File: src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={(movie: Movie) => onAdd(movie)} />
```
Comment (Viktor-Kost) [inline src/App.tsx:22]:
```suggestion
        <NewMovie onAdd={onAdd} />
```

File: src/components/NewMovie/NewMovie.tsx [hunk 3]
```diff
@@ -1,45 +1,94 @@
...
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const initialMovieState = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
```
Comment (Viktor-Kost) [inline src/components/NewMovie/NewMovie.tsx:17]: move outside the Component

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Nice!
Some moments can be improved

**Review by mykhalenych — CHANGES_REQUESTED** [review]
https://github.com/mate-academy/react_movies-list-add-form/assets/53228259/4f702eee-7830-4e32-8bcd-c245832fcb4b

add validation for inputs with url

**Pull request closed without merging**